README.md

```diff
@@ -1,2 +1,171 @@
+Moreover, your public key also needs to be registered and correlated with your app.
+
+It is expected your public key is distributed
+through a [JSON Web Key Set](https://tools.ietf.org/html/rfc7517) endpoint implemented by your backend
+and the only thing you should be sharing is a JWKS endpoint URI.
+
+```
+curl --request GET --url http://your-api.example.com/jwks
+{
+  "keys": [{
+    "alg":"RS256",
+    "kid":"00001",
+    "e":"AQAB",
+    "n":"sZL2ud...SNvqQ",
+    "kty":"RSA"
+  }]
+}
+```
+
+Unlike sharing a static `public.pem` file, this mechanism helps you rotate your keys
+and reduces risk of public keys getting out of sync
+with what has been initially registered within Karhoo auth service.
+
+Having in mind implementing this endpoint may take a while, we prepared a tool which will help
+you instantly publish your public key in the desired format for testing purposes:
+
+```bash
+docker run -d -p 8080:8080 \
+    --name token-exchange \
+    --volume $(pwd)/keys:/keys:ro \
+    karhoo/token-exchange-quickstart
+
+curl --request GET --url http://localhost:8080/jwks > jwks.json
+```
+
+Having a `jwks.json` file created you can now
+host it on your servers as a static file or just share the file with Karhoo
+so that it can be registered as a valid verification key for your app.
+Once it is done you will be provided with a `client_id`
+uniquely identifying your application in the Karhoo platform.
+
+## 3. Issuing tokens
+
+It is time to start issuing tokens.
+JWTs should have the following payload:
+
+```
+{
+  "jti": "b961af11-462e-4685-bca6-c81bf8081cc4",  // Unique token identifier
+  "aud": "https://sso.sandbox.karhoo.com/oauth/v2/token-exchange", // Audience
+  "iss": "your-client-id",         // Client ID as registered within Karhoo auth service
+  "sub": "ecc...f4b",              // The unique identifier of your user
+  "iat": 1578068426,               // A timestamp when the token was issued  
+  "exp": 1578068926,               // A timestamp when the token expires
+  "given_name": "John",            // First name
+  "family_name": "Doe",            // Last name
+  "email": "john.doe@karhoo.com",  // Verified email
+  "phone_number": "+15005550006",  // Verified phone number in E.164 format
+  "locale": "en"   			           // End-User's locale (BCP47 [RFC5646] format)
+}
+```
+
+and should be signed with your private key using `RS256` algorithm.
+
+Since implementing this feature on your backend may take a while, we prepared a tool which will help
+you instantly generate valid JWTs for testing purposes:
+
+```bash
+docker exec -ti token-exchange make jwt
+```
+
+```
+Please provide your user details:
+? User ID c1e30e2c-21b6-40c4-9228-9d940c99174e
+? Given name John
+? Family name Doe
+? Email john.doe@example.com
+? Phone number
+? Locale en
+? Your client ID your-app-id
+eyJra...zRQ7Tn3A
+```
+
+You can now use the output of this command
+as a valid token within Karhoo SDK:
```

Comment:
Is the Karhoo SDK ultimately the only platform that will be supporting it? Otherwise I would mention that it can be used in web flows too.

Reply:
Well, the `Karhoo SDK` in this sense is both `Mobile` and `Web` SDKs.
The mobile is there, the web one not quite (that's why I'm using Kotlin example).

I just want to prevent others from using this for web flow and give them opensourced libraries (like web-lib-auth) in future to achieve this.